commit df567e873326532a96de420e6953d7111f446cff
Author: Baptiste Daroussin <bapt@FreeBSD.org>
Date:   Mon Feb 27 14:31:48 2017 +0100

    Use binary copy instead of printf to fix issues with pkg repo
    signing_command when the signature contains \0

diff --git libpkg/pkg_repo_create.c libpkg/pkg_repo_create.c
index ac69e06c..1dab735c 100644
--- libpkg/pkg_repo_create.c
+++ libpkg/pkg_repo_create.c
@@ -822,7 +822,7 @@ pkg_repo_sign(char *path, char **argv, int argc, UT_string **sig, UT_string **ce
 			break;
 		}
 		if (buf != NULL)
-			utstring_printf(buf, "%s", line);
+			utstring_bincpy(buf, line, linelen);
 	}
 
 	if (pclose(fp) != 0) {
